### DISCUSSION TITLE
ROS2 설치 과정 중 오류

### DISCUSSION THREAD

**Discussion by k2su99** [SKKUAutoLab/H-Mobility-Autonomous-Advanced-Course#84] [2024년도 기타 질문]
안녕하세요. ROS2 설치하는데 첫 명령어부터 이런 오류가 떠서 문의드립니다. 
rufus를 통해 몇번 다시 ubuntu를 설치해보았으나 해결되지 않았습니다. 
구글링 통해 여러방법 시도해보았으나 해결되질 않았습니다. 
해결방법 알려주시면 감사하겠습니다.
![Screenshot from 2024-07-16 17-21-23](https://github.com/user-attachments/assets/d0a486d6-1117-4bb2-84e2-29d98b252c2a)

**Comment by HyeongKeunHong** [comment]
안녕하세요, 혹시 설치하신 우분투 버전이 어떻게 될까요?

**Reply by k2su99** [reply]
영상에 나온대로 22.04 버전으로 설치했습니다!

**Reply by HyeongKeunHong** [reply]
혹시 어떤 명령어를 실행하다가 오류가 났는지 명시해주실 수 있을까요?


첫 명령어라면 아래와 같은 명령을 실행할 때 오류가 나신 것인가요? 
```
locale  # check for UTF-8

sudo apt update && sudo apt install locales
sudo locale-gen en_US en_US.UTF-8
sudo update-locale LC_ALL=en_US.UTF-8 LANG=en_US.UTF-8
export LANG=en_US.UTF-8

locale  # verify settings
```



아니면 아래의 명령어를 실행하다가 오류가 난 것일까요?
```
sudo apt update && sudo apt install curl -y
sudo curl -sSL https://raw.githubusercontent.com/ros/rosdistro/master/ros.key -o /usr/share/keyrings/ros-archive-keyring.gpg
```

**Reply by k2su99** [reply]
위의 명령어를 실행했을 때 해당 오류가 발생했습니다.

**Reply by k2su99** [reply]
아니면 아이디부터 다시 만들 수 있는 우분투 초기화 방법을 알 수 있을까요? 해결이 안되면 아예 처음부터 시도해보려고도 합니다.

**Reply by HyeongKeunHong** [reply]
윈도우로 부팅하신 뒤 하드디스크 파티션 설정창을 여시고, 우분투에 해당하는 볼륨을 삭제하신 뒤, 재설치하시면 됩니다.

**Reply by HyeongKeunHong** [reply]
캡처하신 에러 중간에 보시면 NO_PUBKEY 값이 있습니다. 그 값을 확인하시고 아래의 명령을 수행해보시기 바랍니다.

sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys <<NO_PUBKEY 키입력>>

**Reply by k2su99** [reply]
넵. 늦은 시간에 신경써주셔서 감사합니다! 알려주신 방법대로 해보겠습니다.

**Reply by k2su99** [reply]
말씀해주신 방법으로 ROS2 설치는 완료했습니다. 감사합니다. 그런데 이후 과정 진행중 NVIDIA driver 설치에서 설치가능한 드라이버를 알 수 없는 문제가 생겼습니다. 구글링 통해 최신 버전 설치나 다양한 방법 시도해보았으나 해결되지 않았습니다. 
![스크린샷 2024-07-17 12-40-11](https://github.com/user-attachments/assets/9838f1ed-593a-438d-8466-ee2e75ccbade)

**Reply by HyeongKeunHong** [reply]
이전에 올라왔던 질문 내용과 동일한 현상으로 보여집니다. https://github.com/SKKUAutoLab/H-Mobility-Autonomous-Advanced-Course/discussions/50 에서 해당 내용을 참고하시기 바랍니다.

**Reply by k2su99** [reply]
넵 감사합니다.
